### PR TITLE
Updates for new gs-chunked-io release

### PR DESCRIPTION
**Pull request by xbrianh**

### CODE REVIEW

File: ssds/__init__.py [hunk 1]
```diff
@@ -67,38 +70,44 @@ def upload(self,
...
-                    f = executor.submit(self._upload_oneshot, filepath, ssds_key)
-                    oneshot_futures.add(f)
+        if threads is None:
+            e: Union[contextlib.AbstractContextManager, ThreadPoolExecutor] = contextlib.nullcontext()
+            oneshot_uploads: Optional[AsyncSet] = None
+        else:
+            e = ThreadPoolExecutor(max_workers=threads)
+            oneshot_uploads = AsyncSet(e, concurrency=threads)
+
+        with e:
```
Comment (DailyDreaming) [inline ssds/__init__.py:93]: Nit: "executor" instead of "e".

File: ssds/__init__.py [hunk 2]
```diff
@@ -147,11 +156,12 @@ def _tag():
             tags = {SSDSObjectTag.SSDS_MD5: s3_etag, SSDSObjectTag.SSDS_CRC32C: gs_crc32c}
             self.blobstore.put_tags(self.bucket, key, tags)
 
-        if executor:
-            f = executor.submit(_tag)
-            f.add_done_callback(lambda f: f.result())
-        else:
-            _tag()
+        # if executor:
```
Comment (DailyDreaming) [inline ssds/__init__.py:159]: Saving for later?

File: ssds/cli/staging.py [hunk 3]
```diff
@@ -26,9 +25,8 @@ def upload(args: argparse.Namespace):
     """
     ssds = Staging[args.deployment].ssds
     root = os.path.abspath(os.path.normpath(args.path))
-    with ThreadPoolExecutor(max_workers=4) as e:
-        for ssds_key in ssds.upload(root, args.submission_id, args.name, e):
-            print(ssds.compose_blobstore_url(ssds_key))
+    for ssds_key in ssds.upload(root, args.submission_id, args.name, threads=3):
```
Comment (DailyDreaming) [inline ssds/cli/staging.py:28]: Was `workers` -> `threads` downgraded from 4 to 3?  Should this be a function of the system hardware?

Comment (xbrianh) [reply ssds/cli/staging.py:28]: This is IO bound, not CPU bound. I've noticed that 3 is working better than 4, but I'm not sure how to adjust this number automatically.

File: ssds/__init__.py (hunk 2 above)
Comment (xbrianh) [reply ssds/__init__.py:159]: Will replace with TODO comment

**Comment by xbrianh** [comment]
@mitenjain FYI: this update will have the side-effect of preventing sync failure due to the `Backend Error` exception you previously encountered.

**Pull request merged**